flask_mqtt/__init__.py

```diff
@@ -80,11 +81,12 @@ def init_app(self, app):
         # type: (Flask) -> None
         """Init the Flask-MQTT addon."""
         self.client_id = app.config.get("MQTT_CLIENT_ID", "")
-        self.client = Client(
-            client_id=self.client_id,
-            transport=app.config.get("MQTT_TRANSPORT", "tcp"),
-            protocol=app.config.get("MQTT_PROTOCOL_VERSION", MQTTv311)
-        )
+        if isinstance(self.client_id, unicode):
+            self.client._client_id = self.client_id.encode('utf-8')
```

Comment:
This won't work on Python3 as all Strings are unicode there. Is there a specific reason you want to encode here? Paho MQTT is using normal strings for the client_id: https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py.

Reply:
I encode is the code in paho.mqtt.client.Client __init__ function to set client_id value.
(https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py line 533)
This value is changed apart from __init__ function and i must applied this code part to keep consistency of value.

Reply:
Alright, I see this now. Thanks for the explanation.

Reply:
Still you need to add an alias for `unicode` for Python3 at the top of the file for this to work on Python3.

    # define some alias for python2 compatibility
    if sys.version_info[0] >= 3:
        unicode = str

You wanna add that and I will accept your PR.